.github/workflows/buf-push.yaml

```diff
@@ -0,0 +1,27 @@
+name: buf-push
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: bufbuild/buf-setup-action@v1
+      - uses: bufbuild/buf-lint-action@v1
+      - uses: bufbuild/buf-breaking-action@v1
+        with:
+          # The 'main' branch of the GitHub repository that defines the module.
+          against: "https://github.com/${GITHUB_REPOSITORY}.git#branch=main,ref=HEAD~1"
+      - run: buf generate
+      - run: go mod tidy
+        working-directory: go
+      - run: |
+          git add .
+          git diff --name-only --exit-code HEAD -- .
+      - uses: bufbuild/buf-push-action@v1
+        with:
+          buf_token: ${{ secrets.BUF_TOKEN }}
```

Comment:
This is to publish proto as `buf.build/cncf/xds` in [buf.build registry](https://buf.build/explore)